# Put title here
##### your name

## Abstract

## Introduction

 
## Methods

### RNA-seq analysis
* Describe the datasets you're working with. How many reads were in each RNA-seq experiment? What read lengths?
* Describe and interpret anything flagged by `fastqc`.
* What alignment method and reference genome were used?
* Describe how you used kallisto to quantify transcripts. What non-default parameters did you use and why?
* Describe how you used sleuth to identify differentially expressed transcripts.

## Results

### RNA-seq analysis
* How comparable were the replicates? Which 2 tissues were most similar? Include a figure or table summarizing your pairiwse analysis of each RNA-seq dataset.
* How many diffentially expressed transripts did you find? Include screenshots of several examples from IGV.

### ChIP-seq analysis
* Describe the general signal you see for H3K4me1 and H3K27ac. Where do these marks tend to fall? In introns? beginning of genes? end of genes? Do they look different between the different samples? Hypothesize why. Similarly describe the PhyloP track.
* Describe the pattern of these signals in the ZRS region.

## Discussion


## Citations

