# Put title here
##### your name

## Abstract

## Introduction

## Methods

### RNA-seq analysis
* Describe the datasets you're working with. How many reads were in each RNA-seq experiment? What read lengths?
* Describe and interpret anything flagged by `fastqc`.
* What alignment method and reference genome were used? Include what genome build was used, which is important for enabling other people to reproduce your analysis.
* Describe how you used kallisto to quantify transcripts. What non-default parameters did you use and why?
* Describe how you used sleuth to identify differentially expressed transcripts. What False Discovery Rate (q-value) threshold did you use?

### ChIP-seq and other epigenomic analyses
* Describe the datasets you visualized on IGV. 
* What tools did you use for the multiple sequence alignment and visualization? Mention any non-default parameters.
* What tool and database did you use for motif identification.

## Results

### RNA-seq analysis
* How comparable were the replicates? Which 2 tissues were most similar? Include a figure or table summarizing your pairiwse analysis of each RNA-seq dataset.
* How many diffentially expressed transripts did you find? Include screenshots of several examples from IGV.

### ChIP-seq analysis
* Describe the general signal you see for H3K4me1 and H3K27ac. Where do these marks tend to fall? In introns? beginning of genes? end of genes? Do they look different between the different samples? Hypothesize why. Similarly describe the PhyloP track.
* Describe the pattern of these signals in the ZRS region.
* Provide a figure for the MSA results. Are there any sequences missing in snakes that are present in all other animals analyzed?
* Provide a table of the top matching motifs from part 9.

## Discussion
* Summarize your analysis of the ZRS region and your interpretation of its involvement in determining whether an organism develops limbs or not.
* We identified the ZRS region based on knowledge from previous literature. Are there other enhancers that appear to be both limb-specific an highly conserved nearby Shh? If you wanted to identify these regions computationally how would you do it?
* Do you have any hypotheses about why the identified mutations might lead to a loss of legs in snakes? What further experiments or analyses could be done to determine the function of this sequence?

## Citations

